src/bin/pgcopydb/ld_stream.c

```diff
@@ -490,7 +491,7 @@ startLogicalStreaming(StreamSpecs *specs)
 				  OutputPluginToString(specs->slot.plugin),
 				  specs->pluginOptions.count);
 
-		if (!pgsql_start_replication(&stream))
+		if (!pgsql_start_replication(&stream, specs->sourceDB, specs->paths.dir))
```

Comment:
The whole idea behing the latest opened discussion in this review is to allow to skip using the sourceDB when doing `pgsql_start_replication()`, getting back to your point about separation of concerns. At replication start, only store the timeline history to a file.

In `stream_write_context()` (or maybe in a new function, if needed?) we can then parse the history file with a file iterator and store our own format in the SQLite catalogs.